pages/blog/typescript-types.mdx

```diff
@@ -0,0 +1,153 @@
+
+### Operator `typeof`
+
+Varibale of a union type can assume one or even few diffrent types. In the example below we use `if` statement to check if the variable is of type `Food` or not.
+We can also use the opretor `typeof` with the variable name and compare it with the type you expect for the variable.
+Lets see a basic example of `typeof` operator in function:
+
+```ts
+function getScoreofLovingPizza(score: number | 'very-much') {
+  if (typeof score === 'very-much') {
+    let score = 10
+    return console.log(`How much the use love pizza - score number: ${score}`)
+  } else {
+    return console.log(`How much the use love pizza - score number: ${score}`)
+  }
+}
+```
+
+[Here's a live example in TS Playground](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAcwKZQMoTgJ1XYAGTgDcYxkAFGALxoEMAKAZ2zwC5EwQBbAI1Q5EAH0QByEoICeAWh4gIACzEBKRAG8AsAChEiGMEQs2qRAF4L4yTlnylqjTr168UEDiQBGAAwBuJ4gAvoioADbMplq6zq7uSKy4qAGBOinaQA)
+
+### TypeScript Type Guard
+
+Type Guard is a conditional statement that evaluates the type of a variable and if it is of the expected type, it will execute the code block.
+It can be implemented using the `typeof` operator like we learn in the previous example. We can also use type guard with `in` operator to check if the variable is of the expected type.
+
+```ts
+interface Sushi {
+  fish: number
+}
+interface Pizza {
+  topping: string
+}
+
+function identifyFood(food: Sushi | Pizza) {
+  if ('fish' in food) {
+    food
+    // ^?
+  } else {
+    food
+    // ^?
+  }
+}
+```
+
+[Here's a live example in TS Playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMoFcDOALYyDeAsAFDLIzA4BcyI6AtgEbQDcJAviaJLIigArAAXkLgESZMAHsADjNABzGpjBRFbYp2IkY6EAjDApIZMAAmEcMBgBPAGJSpZgBQxHZmhhx4APskEicACU4qSmMMjOAOQUOFGmJm5OIURhZElmGmRkAPQ5yAB6APwSyOzIEAA2mCip2eTuWdl5hSVhWuwsQA)
+
+## Conclusion
+
+- I hope this blog post help you better the why of TypeScript types system work and how to use it for your projects.
+
+- In this brief blog post, we looked at a few feature of TypeScript that help you on code-time, mosthly help you to understand the output of any function you create.
+
+- All examples above allows us to build more versatile types by relying on a template pattern for the discriminant property rather than an exact string.
+
+- Recently i beed use GraphQL Codegen plugin to generate my types from my GraphQL schema. When my project is small, as a developer i can create and decalre any type that i want. But when the project is big, it hard to manage the types becuse we have too much types to create.
+  So the best way to manage the types, in my openion, is to use GraphQL Codegen plugin.
+
+## So how do we creating Types based GraphQL schema?
```

Comment:
I think this needs more background:

* GraphQL also has `union` and `interface` -> explain how it connected to the TS feature
* How GraphQL Codegen can help you -> add example